native_client/alphabet.h

```diff
@@ -0,0 +1,50 @@
+#ifndef ALPHABET_H
+#define ALPHABET_H
+
+#include <cassert>
+#include <fstream>
+#include <string>
+#include <unordered_map>
+
+/*
+ * Loads a text file describing a mapping of labels to strings, one string per
+ * line. This is used by the decoder, client and Python scripts to convert the
+ * output of the decoder to a human-readable string and vice-versa.
+ */
+class Alphabet {
+public:
+  Alphabet(const char *config_file) {
+    std::ifstream in(config_file, std::ios::in);
+    unsigned int label = 0;
+    for (std::string line; std::getline(in, line);) {
+      if (line[0] == '#') {
+        continue;
+      }
+      label_to_str_[label] = line;
+      str_to_label_[line] = label;
+      ++label;
+    }
+    size_ = label;
+    in.close();
+  }
+
+  const std::string& StringFromLabel(unsigned int label) {
```

Comment:
nit - This method is not `const`, as `operator[]` is not `const`, however this method should be `const`.

If a `label` is passed in that is not already a key in `label_to_str_`  an empty `string` will be added to `label_to_str_` and returned. To avoid this, unfortunately, you should use the `find`method of `label_to_str_`

Reply:
Nice catch!

Reply:
Although, because of the assert below that should never happen. I'll still make it a const method though.